### PR TITLE
Fix linker error LNK2005 for tag dispatching staff

### PR DESCRIPTION
**Pull request by SergeyKopienko**
In this PR we fix linker error when we have more then one translation unit for one project:
```C++
error LNK2005: "class oneapi::dpl::execution::v1::parallel_policy const & __cdecl oneapi::dpl::get_unvectorized_policy(class oneapi::dpl::execution::v1::parallel_unsequenced_policy const &)" (?get_unvectorized_policy@dpl@oneapi@@YAAEBVparallel_policy@v1@execution@12@AEBVparallel_unsequenced_policy@4512@@Z) already defined in hello-1-5be217.obj
```

The problem is that our functions `get_unvectorized_policy` are not `inline`.

### CODE REVIEW

File: include/oneapi/dpl/pstl/glue_memory_impl.h [hunk 1]
```diff
@@ -187,13 +187,13 @@ uninitialized_fill_n(_ExecutionPolicy&& __exec, _ForwardIterator __first, _Size
 
 #if (_PSTL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN || _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN)
 
-const oneapi::dpl::execution::parallel_policy&
+inline const oneapi::dpl::execution::parallel_policy&
```
Comment (rarutyun) [inline include/oneapi/dpl/pstl/glue_memory_impl.h:190]: I hope we don't have a similar situation with `__select_backend`, do we?

Comment (SergeyKopienko) [reply include/oneapi/dpl/pstl/glue_memory_impl.h:190]: The `__select_backend` functions is template functions so there are no problems for them.

**Review by Alexandr-Konovalov — APPROVED** [review]
LGTM for the fix, but I don't now current policy wrt the regression testing.

File: include/oneapi/dpl/pstl/glue_memory_impl.h (hunk 1 above)
Comment (rarutyun) [reply include/oneapi/dpl/pstl/glue_memory_impl.h:190]: That's what I was asking for. Thanks!

**Review by rarutyun — APPROVED** [review]

**Pull request merged**